sentry/src/main/java/io/sentry/TransactionContext.java

```diff
@@ -0,0 +1,56 @@
+package io.sentry;
+
+import io.sentry.protocol.SentryId;
+import io.sentry.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class TransactionContext extends SpanContext {
+  private final @NotNull String name;
+  private @Nullable Boolean parentSampled;
+
+  /**
+   * Creates {@link TransactionContext} from sentry-trace header.
+   *
+   * @param name - the transaction name
+   * @param sentryTrace - the sentry-trace header
+   * @return the transaction contexts
+   */
+  public static @NotNull TransactionContext fromSentryTrace(
```

Comment:
fwiw on top of fromSentryTrace/toSentryTrace I think it would be good if we had an implementation of `continueFromHeaders` and `iterHeaders`. Those are the actual APIs used by integrations as then integrations don't need updates if we decided to write more headers (which will happen with dynamic sampling)

Proposal here: https://github.com/getsentry/develop/pull/204